skfda/misc/feature_construction/_functional_transformers.py

```diff
@@ -0,0 +1,508 @@
+    at :math:`t_0 > 0` if for some :math:`\epsilon >0`, X(t) $\leq$
+    c if t :math:'\in (t_0 - \epsilon, t_0) and X(t) $\geq$ c if
+    :math:`t\in (t_0, t_0+\epsilon)`.
+
+    If the trajectories are differentiable, then
+    :math:`N_i = card\{t \in[a,b]: X(t) = c_i, X' (t) > 0\}.`
+
+        Args:
+            data: FDataGrid where we want to calculate
+            the number of up crossings.
+            intervals: sequence of tuples containing the
+            intervals we want to consider for the crossings.
+        Returns:
+            ndarray of shape (n_dimensions, n_samples)\
+            with the values of the counters.
+
+    Example:
+
+    We import the Phoneme dataset and for simplicity we use
+    the first 200 samples.
+    >>> from skfda.datasets import fetch_phoneme
+    >>> dataset = fetch_phoneme()
+    >>> X = dataset['data'][:200]
+
+    Then we decide the interval we want to consider (in our case (5.0,7.5))
+    and call the function with the dataset. The output will be the number of
+    times each curve cross the interval (5.0,7.5) growing.
+    >>> from skfda.misc.feature_construction import number_up_crossings
+    >>> number_up_crossings(X, [(5.0,7.5)])
+    array([[1, 20, 69, 64, 42, 33, 14, 3, 35, 31, 0, 4, 67, 6, 12, 16, 22, 1,
+            25, 30, 2, 27, 61, 0, 11, 20, 3, 36, 28, 1, 67, 36, 12, 29, 2,
+            16, 25, 1, 24, 57, 65, 26, 20, 18, 43, 0, 35, 40, 0, 2, 56, 4,
+            21, 28, 1, 0, 19, 24, 1, 2, 8, 63, 0, 2, 3, 3, 0, 8, 3, 2, 10,
+            62, 72, 19, 36, 46, 0, 1, 2, 18, 1, 10, 67, 60, 20, 21, 23, 12,
+            3, 30, 21, 1, 57, 64, 15, 4, 4, 17, 0, 2, 31, 0, 5, 24, 56, 8,
+            11, 14, 17, 1, 25, 1, 3, 61, 10, 33, 17, 1, 12, 18, 0, 2, 57, 4,
+            6, 5, 2, 0, 7, 17, 4, 23, 60, 62, 2, 19, 21, 0, 42, 28, 0, 10,
+            29, 74, 34, 29, 7, 0, 25, 23, 0, 15, 19, 1, 43, 1, 11, 9, 4, 0,
+            0, 2, 1, 54, 55, 14, 14, 6, 1, 24, 20, 2, 27, 55, 62, 32, 26, 24,
+            37, 0, 26, 28, 1, 3, 41, 64, 8, 6, 27, 12, 1, 5, 16, 0, 0, 61,
+            62, 1, 3, 7]], dtype=object)
+    """
+    curves = data.data_matrix
+    if curves.shape[2] != len(intervals):
+        raise ValueError(
+            "Sequence of intervals should have the "
+            + "same number of dimensions as the data samples",
+        )
+    transformed_counters = []
+    for index, interval in enumerate(intervals):
+        a, b = interval
+        if b < a:
+            raise ValueError(
+                "Interval limits (a,b) should satisfy a <= b. "
+                + str(interval) + " doesn't",
+            )
+        curves_counters = []
+        for c in curves:
+            counter = _n_curve_crossings(c, index, interval)
+            curves_counters = curves_counters + [counter]
+
+        transformed_counters = transformed_counters + [curves_counters]
+
+    return np.asarray(transformed_counters, dtype=list)
+
+
+def moments_of_norm(
```

Comment:
I don't know of these last two should be included, and if they are correctly implemented. In any case, they should had been vectorized.